### PR TITLE
Add more detailed error handling to doc-upload

### LINKED ISSUE #62: cargo doc-upload reports "already up-to-date" after auth failure
Hey, thanks for making such a cool cargo plugin. We use it at [tantivy](https://github.com/tantivy-search/tantivy/) to auto-update docs for the searcher/indexer library that we are developing. 

I was reviewing the [travis log from our recent build](https://api.travis-ci.org/v3/job/544505108/log.txt) and noticed an inconsistency between the authentication error and the last thing printed by doc-upload

```bash
remote: Invalid username or password.
fatal: Authentication failed for 'https://[secure]@github.com/tantivy-search/tantivy.git/'
Documentation already up-to-date.
```

I think the last line is from

https://github.com/roblabla/cargo-travis/blob/320fcac953b17e9a19dd0515216ae7a9fd2da36e/src/lib.rs#L367

The if/else logic implies that if documentation wasn't successfully updated, it must be already up to date, instead of reporting an error.

In our case, we had an authentication error, which prevented us from updating documentation. 

I think we should parse and log the exit status of the command, so logs can show people the real reason for failure. 

Will it help if I send a PR or are you too busy to review?

### PR DESCRIPTION
**Pull request by petr-tik**
Use output + stdout instead of status.

Match on return code. If it's not 0 search for key words in stdout and summarise
the error to the user

closes #62 - bumped version +1

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -371,20 +371,40 @@ pub fn doc_upload(message: &str, origin: &str, gh_pages: &str, doc_path: &str, l
...
+                        if output_str.contains("invalid username or password") {
+                            println!("Invalid username or password - make sure TravisCI user can push to {} branch on your repo", gh_pages);
+                        }
+                    } else {
+                        println!("{}", output_str);
+                    }
+                }
+            }
+            None => {
+                println!("Process terminated by signal and never returned an exit code");
```
Comment (CAD97) [inline src/lib.rs]: I'd write this without assuming signals as the OS implementation:

> Git terminated without reporting an error code

**Comment by roblabla** [comment]
Please don't try to parse the human-readable output of git. There are 0 stability guarantees around them, and it makes for incredibly fragile code.

Instead, let the output of git flow to stdout (like it did before, `.status()` does that by default), and simply wrap the call in `require_success`. This way, if git returns an error, `cargo doc-upload` will exit with an error as well - which is what I meant by bubbling.

**Comment by petr-tik** [comment]
Thanks for your swift review. 

As far as I understood it, @roblabla's comment suggests wrapping the call in `require_success`, which pretty much overrules @CAD97 comments inline. 

I reverted to using `status()`, which already pipes git's stdout to io::Stdout. I also took the liberty to cargo-fmt the dependencies. My editor also cargo-fmt'ed the rest of the file, but I wanted to keep the revision easy to review. 

Let me know, if something else is missing. 

If it's all good to merge, feel free to squash into 1 commit before merging.

**Comment by roblabla** [comment]
Thanks for the PR! Will release version 0.0.11 in a minute.

**Pull request merged**